client/components/LearnerViewDirective.js

```diff
@@ -477,10 +477,22 @@ tie.directive('learnerView', [function() {
             feedbackDiv.scrollTop = feedbackDiv.scrollHeight +
               additionalHeightForLoadingIndicator;
             $timeout(function() {
-              SolutionHandlerService.processSolutionAsync(
-                tasks[currentTaskIndex], code,
+              var index = 0;
```

Comment:
Right. How many secs/millisecs is it in practice? Just wanted to get a feel for that.

Re your first paragraph, yes, that was my thought -- find some way to batch all the test code together and send it to Skulpt for a single exec run. Open to debate on the pros/cons of that alternative vs this one, just want to make sure you've considered it.